sw/device/silicon_creator/lib/drivers/pinmux.c

```diff
@@ -19,6 +20,7 @@ enum {
    * Base address of the pinmux registers.
    */
   kBase = TOP_EARLGREY_PINMUX_AON_BASE_ADDR,
+  kPadAttrSpinWaitCycles = 500,  // 500 cycles is 5us with a 100MHz clock
```

Comment:
nit: we can actually test this assumption in a unit test using the device constants.

Reply:
Hmm, not sure I follow, wouldn't this be better checked with a static assert? Another issue to consider is that the system clock is different different for different platforms, so the `kClockFreqCpuHz` values in the device_*.c files are different for FPGA vs DV (which reflect Si). Additionally, this value is sort of arbitrary and could change if we used a different node / pad library I think.

Reply:
> Hmm, not sure I follow, wouldn't this be better checked with a static assert?

Unless the driver already depends on those device constants I'd prefer adding a simple test to the already existing unit test.

> Another issue to consider is that the system clock is different different for different platforms, so the `kClockFreqCpuHz` values in the device_*.c files are different for FPGA vs DV (which reflect Si). 

Correct, but IIUC this only matters for the actual chip, so we would need to check that 500 cycles >= 5 us at the expected freq.

> Additionally, this value is sort of arbitrary and could change if we used a different node / pad library I think.

Sure, in that case we would either use the worst case value for the driver or, if needed, have different constants that could be checked separately.

BTW, I don't think this is a super critical issue for bootstrap and RMA because both of them use strong pull-ups/downs to simplify sampling.

Reply:
Sg, I _think_ I have implemented what you are saying above in #18020 (I added another commit to the PR). Lmk what you think.